src/pages/view/TitleBar/FilterDialog.js

```diff
@@ -90,6 +121,9 @@ const FilterDialog = ({
                     <PeriodDimension
                         selectedPeriods={selectedItems}
                         onSelect={commonProps.onSelect}
+                        excludedPeriodTypes={getExcludedPeriodTypes(
```

Comment:
Is the idea with making excludedPeriodTypes configurable in PeriodDimension that the app may want to be even more restrictive than the system settings? Could the app also enforce a less restrictive list than the system settings?

Reply:
In my initial version of this, the fetching of system settings and excluding of periods were all done in Analytics itself. But it was decided to move it out to `excludedPeriodTypes` and let the apps decide which periods to exclude, since we want to keep the lib flexible and unopinionated. Also, since the apps already fetch the system settings, we avoid having a separate backend call in Analytics which would also lead to a having a loading spinner when you open the period dimension. 
So by letting the apps fetch the settings (which the already do, these are just additional props to the existing api request), we avoid both the loading spinner, an additional api request and oppinionating Analytics.

So to answer the question, yes apps could provide any restrictions they want, regardless of the system settings.

Reply:
ok, if that's approved by the product manager, then ok with me.